google-cloud-pubsublite/src/test/java/com/google/cloud/pubsublite/internal/TopicBacklogReaderImplTest.java

```diff
@@ -0,0 +1,189 @@
+    ApiFuture<ComputeMessageStatsResponse> future =
+        reader.computeMessageStats(
+            builder
+                .put(Partition.of(1), Offset.of(10))
+                .put(Partition.of(2), Offset.of(20))
+                .build());
+
+    ExecutionException ex = assertThrows(ExecutionException.class, future::get);
+    assertThat(ExtractStatus.extract(ex.getCause()).get().getCode()).isEqualTo(Code.UNAVAILABLE);
+  }
+
+  @Test
+  public void computeMessageStats_aggregatesEmptyMessages() throws Exception {
+    ComputeMessageStatsResponse partition1 = ComputeMessageStatsResponse.getDefaultInstance();
+    ComputeMessageStatsResponse partition2 = ComputeMessageStatsResponse.getDefaultInstance();
+    ComputeMessageStatsResponse aggregate = ComputeMessageStatsResponse.getDefaultInstance();
+
+    when(mockClient.computeMessageStats(
+            topicPath, Partition.of(1), Offset.of(10), Offset.of(Integer.MAX_VALUE)))
+        .thenReturn(ApiFutures.immediateFuture(partition1));
+    when(mockClient.computeMessageStats(
+            topicPath, Partition.of(2), Offset.of(20), Offset.of(Integer.MAX_VALUE)))
+        .thenReturn(ApiFutures.immediateFuture(partition2));
+
+    ImmutableMap.Builder<Partition, Offset> builder = ImmutableMap.builder();
+    ApiFuture<ComputeMessageStatsResponse> future =
+        reader.computeMessageStats(
+            builder
+                .put(Partition.of(1), Offset.of(10))
+                .put(Partition.of(2), Offset.of(20))
+                .build());
+
+    assertThat(future.get()).isEqualTo(aggregate);
+  }
+
+  @Test
+  public void computeMessageStats_timestampsAggregatedWhenPresent() throws Exception {
+    Timestamp minEventTime = Timestamp.newBuilder().setSeconds(1000).setNanos(10).build();
+    Timestamp minPublishTime = Timestamp.newBuilder().setSeconds(1001).setNanos(11).build();
+    ComputeMessageStatsResponse partition1 =
+        ComputeMessageStatsResponse.newBuilder().setMinimumPublishTime(minPublishTime).build();
+    ComputeMessageStatsResponse partition2 =
+        ComputeMessageStatsResponse.newBuilder().setMinimumEventTime(minEventTime).build();
+    ComputeMessageStatsResponse aggregate =
+        ComputeMessageStatsResponse.newBuilder()
+            .setMinimumEventTime(minEventTime)
+            .setMinimumPublishTime(minPublishTime)
+            .build();
+
+    when(mockClient.computeMessageStats(
+            topicPath, Partition.of(1), Offset.of(10), Offset.of(Integer.MAX_VALUE)))
+        .thenReturn(ApiFutures.immediateFuture(partition1));
+    when(mockClient.computeMessageStats(
+            topicPath, Partition.of(2), Offset.of(20), Offset.of(Integer.MAX_VALUE)))
+        .thenReturn(ApiFutures.immediateFuture(partition2));
+
+    ImmutableMap.Builder<Partition, Offset> builder = ImmutableMap.builder();
+    ApiFuture<ComputeMessageStatsResponse> future =
+        reader.computeMessageStats(
+            builder
```

Comment:
Just use ImmutableMap.of

Reply:
Oh, didn't know that existed - much better